decidim-proposals/app/cells/decidim/proposals/proposal_m_cell.rb

```diff
@@ -132,7 +132,7 @@ def cache_hash
         hash << model.cache_key_with_version
         hash << model.proposal_votes_count
         hash << model.endorsements_count
-        hash << Digest::MD5.hexdigest(model.component.settings.to_json)
```

Comment:
Could you check what `component.settings` does? Is it taking into account step changes in a multi-step process, for example? That is, a process with multiple steps, and try activating one or another.

If `component.settings` returns the same value then it's fine, but if it changes then we'd need to keep the previous code because buttons ("Vote") might change...

Reply:
I have locally a process with two steps, first of these active. In that process, I have a proposals component with ID 3 (from seed data).

This is what I have before I change the step:
```
irb(main):005:0> Decidim::Component.find(3).settings
  Decidim::Component Load (0.9ms)  SELECT  "decidim_components".* FROM "decidim_components" WHERE "decidim_components"."id" = $1 ORDER BY "decidim_components"."weight" ASC, "decidim_components"."manifest_name" ASC LIMIT $2  [["id", 3], ["LIMIT", 1]]
  Decidim::ParticipatoryProcess Load (1.0ms)  SELECT  "decidim_participatory_processes".* FROM "decidim_participatory_processes" WHERE "decidim_participatory_processes"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  Decidim::Organization Load (1.0ms)  SELECT  "decidim_organizations".* FROM "decidim_organizations" WHERE "decidim_organizations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
=> #<#<Class:0x0000564b4add26b0>:0x0000564b49e7f1b8 @default_locale="en", @scope_id=nil, @vote_limit=0, @announcement={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_limit=0, @scopes_enabled=false, @announcement_ca="", @announcement_en="", @announcement_es="", @proposal_length=500, @allow_card_image=true, @comments_enabled=true, @geocoding_enabled=true, @amendments_enabled=true, @attachments_allowed=true, @comments_max_length=0, @minimum_votes_per_user=0, @new_proposal_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @threshold_per_proposal=0, @new_proposal_help_text_ca="", @new_proposal_help_text_en="", @new_proposal_help_text_es="", @new_proposal_body_template={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @official_proposals_enabled=true, @proposal_answering_enabled=true, @amendments_wizard_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @participatory_texts_enabled=false, @collaborative_drafts_enabled=true, @proposal_edit_before_minutes=5, @new_proposal_body_template_ca="", @new_proposal_body_template_en="", @new_proposal_body_template_es="", @resources_permissions_enabled=true, @amendments_wizard_help_text_ca="", @amendments_wizard_help_text_en="", @amendments_wizard_help_text_es="", @proposal_wizard_step_1_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_2_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_3_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_4_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_1_help_text_ca="", @proposal_wizard_step_1_help_text_en="", @proposal_wizard_step_1_help_text_es="", @proposal_wizard_step_2_help_text_ca="", @proposal_wizard_step_2_help_text_en="", @proposal_wizard_step_2_help_text_es="", @proposal_wizard_step_3_help_text_ca="", @proposal_wizard_step_3_help_text_en="", @proposal_wizard_step_3_help_text_es="", @proposal_wizard_step_4_help_text_ca="", @proposal_wizard_step_4_help_text_en="", @proposal_wizard_step_4_help_text_es="", @can_accumulate_supports_beyond_threshold=false, @proposal_edit_time="limited">
```

I change the step and this is what I get:
```
irb(main):006:0> Decidim::Component.find(3).settings
  Decidim::Component Load (1.4ms)  SELECT  "decidim_components".* FROM "decidim_components" WHERE "decidim_components"."id" = $1 ORDER BY "decidim_components"."weight" ASC, "decidim_components"."manifest_name" ASC LIMIT $2  [["id", 3], ["LIMIT", 1]]
  Decidim::ParticipatoryProcess Load (0.9ms)  SELECT  "decidim_participatory_processes".* FROM "decidim_participatory_processes" WHERE "decidim_participatory_processes"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  Decidim::Organization Load (1.1ms)  SELECT  "decidim_organizations".* FROM "decidim_organizations" WHERE "decidim_organizations"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
=> #<#<Class:0x0000564b4add26b0>:0x0000564b49b2fc88 @default_locale="en", @scope_id=nil, @vote_limit=0, @announcement={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_limit=0, @scopes_enabled=false, @announcement_ca="", @announcement_en="", @announcement_es="", @proposal_length=500, @allow_card_image=true, @comments_enabled=true, @geocoding_enabled=true, @amendments_enabled=true, @attachments_allowed=true, @comments_max_length=0, @minimum_votes_per_user=0, @new_proposal_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @threshold_per_proposal=0, @new_proposal_help_text_ca="", @new_proposal_help_text_en="", @new_proposal_help_text_es="", @new_proposal_body_template={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @official_proposals_enabled=true, @proposal_answering_enabled=true, @amendments_wizard_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @participatory_texts_enabled=false, @collaborative_drafts_enabled=true, @proposal_edit_before_minutes=5, @new_proposal_body_template_ca="", @new_proposal_body_template_en="", @new_proposal_body_template_es="", @resources_permissions_enabled=true, @amendments_wizard_help_text_ca="", @amendments_wizard_help_text_en="", @amendments_wizard_help_text_es="", @proposal_wizard_step_1_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_2_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_3_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_4_help_text={"ca"=>"", "en"=>"", "es"=>"", :ca=>"", :en=>"", :es=>""}, @proposal_wizard_step_1_help_text_ca="", @proposal_wizard_step_1_help_text_en="", @proposal_wizard_step_1_help_text_es="", @proposal_wizard_step_2_help_text_ca="", @proposal_wizard_step_2_help_text_en="", @proposal_wizard_step_2_help_text_es="", @proposal_wizard_step_3_help_text_ca="", @proposal_wizard_step_3_help_text_en="", @proposal_wizard_step_3_help_text_es="", @proposal_wizard_step_4_help_text_ca="", @proposal_wizard_step_4_help_text_en="", @proposal_wizard_step_4_help_text_es="", @can_accumulate_supports_beyond_threshold=false, @proposal_edit_time="limited">
```

I save the first one to `before.txt` and the second one to `after.txt`. I run a diff against those (I have removed the class reference e.g. `#<#<Class:0x0000564b4add26b0>:0x0000564b49b2fc88 ` from the saved files because they are obviously different):
```
$ diff before.txt after.txt 
$
```

This indicates there are no changes in the component settings when I change the step.

I think the component settings are only changed when the component is updated:
https://github.com/decidim/decidim/blob/53b3067ceeeef8de8477c4372cd24cca9858c9a9/decidim-admin/app/commands/decidim/admin/update_component.rb#L42

And every time the component is updated, the `updated_at` column changes which is used to indicate the cache version.

Even if component settings are updated outside of the component admin panel view, I would guess most of the times it's handled through `component.update!(...)` or `component.settings = settings ; component.save!` which both will change the `updated_at` column. I think so because `settings` is a database column for the component, so it has to be updated through the component.

Of course, it is technically possible to change the component settings also in other ways such as `Decidim::Component.update_all(...)` or `component.update_attribute(:settings, {...})` but both of these would cause a linter error (`Rails/SkipsModelValidations`).

Reply:
Nice, thanks!

Reply:
@mrcasals Note that it is anyways an actual bug what you described but it's not caused by this PR. You have probably already noticed I filed a bug for this (#7600).